# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by scripts/Create-CopyPatch.
# 
# T2 SDE: package/*/linux/12-conf-hacks.patch
# Copyright (C) 2004 - 2021 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Set all unset options to module. Needed by the T2 SDE Linux kernel
auto configuration.

Initally written by Clifford Wolf <clifford@clifford.at> and adapted
for various new 2.5, 2.6, 3.x, 4.x and 5.x kernels
by Rene Rebe <rene@exactcode.de>.

--- ./scripts/kconfig/conf.c.orig	2021-06-28 00:21:11.000000000 +0200
+++ ./scripts/kconfig/conf.c	2021-06-29 18:16:43.215860970 +0200
@@ -23,6 +23,7 @@
 	oldaskconfig,
 	syncconfig,
 	oldconfig,
+	no2modconfig,
 	allnoconfig,
 	allyesconfig,
 	allmodconfig,
@@ -166,6 +167,7 @@
 	def_y2m,
 	def_m2y,
 	def_no,
+	def_no2m,
 	def_random
 };
 
@@ -228,8 +230,23 @@
 	}
 
 	for_all_symbols(i, sym) {
-		if (sym_has_value(sym) || sym->flags & SYMBOL_VALID)
+		if (sym_has_value(sym)) /* user defined? */
 			continue;
+
+                if ((mode == def_no2m) && (sym->type == S_TRISTATE)) {
+		       fprintf(stderr, "%s %d\n", sym->name, sym_get_tristate_value(sym));
+                      if (sym_get_tristate_value(sym) != mod)
+                              fprintf(stderr, "Setting %s to 'm'.\n", sym->name);
+                      if (sym_get_tristate_value(sym) != mod) {
+                              sym->def[S_DEF_USER].tri = mod;
+                              sym->flags |= SYMBOL_DEF_USER;
+		       }
+                      continue;
+                }
+
+		if (sym->flags & SYMBOL_VALID)
+			continue;
+
 		switch (sym_get_type(sym)) {
 		case S_BOOLEAN:
 		case S_TRISTATE:
@@ -690,6 +705,7 @@
 	{"allyesconfig",  no_argument,       &input_mode_opt, allyesconfig},
 	{"allmodconfig",  no_argument,       &input_mode_opt, allmodconfig},
 	{"alldefconfig",  no_argument,       &input_mode_opt, alldefconfig},
+	{"no2modconfig",  no_argument,       &input_mode_opt, no2modconfig},
 	{"randconfig",    no_argument,       &input_mode_opt, randconfig},
 	{"listnewconfig", no_argument,       &input_mode_opt, listnewconfig},
 	{"helpnewconfig", no_argument,       &input_mode_opt, helpnewconfig},
@@ -799,6 +815,7 @@
 	case olddefconfig:
 	case yes2modconfig:
 	case mod2yesconfig:
+	case no2modconfig:
 		conf_read(NULL);
 		break;
 	case allnoconfig:
@@ -851,6 +867,9 @@
 	}
 
 	switch (input_mode) {
+	case no2modconfig:
+		conf_set_all_new_symbols(def_no2m);
+		break;
 	case allnoconfig:
 		conf_set_all_new_symbols(def_no);
 		break;
--- linux-5.13/scripts/kconfig/Makefile.orig	2021-06-28 00:21:11.000000000 +0200
+++ linux-5.13/scripts/kconfig/Makefile	2021-06-29 17:15:30.664703774 +0200
@@ -69,7 +69,7 @@
 #  deprecated for external use
 simple-targets := oldconfig allnoconfig allyesconfig allmodconfig \
 	alldefconfig randconfig listnewconfig olddefconfig syncconfig \
-	helpnewconfig yes2modconfig mod2yesconfig
+	helpnewconfig yes2modconfig mod2yesconfig no2modconfig
 
 PHONY += $(simple-targets)
 
@@ -127,6 +127,7 @@
 	@echo  '                    except those preserved by LMC_KEEP environment variable'
 	@echo  '  defconfig	  - New config with default from ARCH supplied defconfig'
 	@echo  '  savedefconfig   - Save current config as ./defconfig (minimal config)'
+	@echo  '  no2modconfig    - New config selecting modules for disabled options'
 	@echo  '  allnoconfig	  - New config where all options are answered with no'
 	@echo  '  allyesconfig	  - New config where all options are accepted with yes'
 	@echo  '  allmodconfig	  - New config selecting modules when possible'
